app/auth.go

```diff
@@ -142,24 +163,54 @@ func (m *MenderAuthManager) MakeAuthRequest() (*client.AuthRequest, error) {
 	}, nil
 }
 
-func (m *MenderAuthManager) RecvAuthResponse(data []byte) error {
+func (m *MenderAuthManager) RecvAuthResponse(data []byte, tenantToken string, serverURl string) error {
 	if len(data) == 0 {
 		return errors.New("empty auth response data")
 	}
 
 	if err := m.store.WriteAll(datastore.AuthTokenName, data); err != nil {
 		return errors.Wrapf(err, "failed to save auth token")
 	}
+	if err := m.store.WriteAll(datastore.AuthTenantTokenName, []byte(tenantToken)); err != nil {
+		return errors.Wrapf(err, "failed to save tenant token")
+	}
+	if err := m.store.WriteAll(datastore.AuthServerURLName, []byte(serverURl)); err != nil {
+		return errors.Wrapf(err, "failed to save serverURL token")
+	}
```

Comment:
Since these values belong together, please use transactions to write them. A good example is how `WriteAll` itself is [implemented](https://github.com/mendersoftware/mender/blob/master/store/dbstore.go#L106-L108). Basically just make sure all the writes are inside one transaction.